src/Cli/dotnet/commands/dotnet-workload/install/MsiInstallerBase.cs

```diff
@@ -285,17 +285,21 @@ protected uint RepairMsi(string productCode, string logFile)
         /// <exception cref="InvalidOperationException"></exception>
         protected void UpdateInstallMode(SdkFeatureBand sdkFeatureBand, bool newMode)
         {
+            string path = Path.Combine(WorkloadInstallType.GetInstallStateFolder(sdkFeatureBand, DotNetHome), "default.json");
+            var installStateContents = File.Exists(path) ? InstallStateContents.FromString(File.ReadAllText(path)) : new InstallStateContents();
```

Comment:
Can we reduce duplication with something like (code seems to repeat a few times)

```
GetInstallStateContents(string path) =>
    InstallStateContents.FromString(File.ReadAllText(path)) : new InstallStateContents();
```